### PR TITLE
#159303925 An Endpoint To Edit A Person

### PR DESCRIPTION
**Pull request by GbengaOshinaga**
```
{
	"name": "John",
	"role": "Operations Manager"
}
```
>- You should receive the following data:
```
{
    "status": "success",
    "message": "Details for John updated successfully",
    "data": {
        "name": "John",
        "image_url": null,
        "role": "Operations Manager",
        "id": "-LIuX9JHf77Hi_XGvRSL",
        "email": "gbenga@andela.com"
    }
}
```

### CODE REVIEW

File: seeders/seed_center.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
+"""Module to seed centers table"""
```
Comment (Victoradukwu) [inline seeders/seed_center.py]: You may want to sync with @ibutiti so both you agree on the same implementation of center seeder.

File: seeders/seed_user.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            name=data.get('name'),
+            email=data.get('email'),
+            center_id=data.get('center_id'),
+            role_id=data.get('role_id')) for data in user_data
+    ]
+
+    for user in users:
+        user.save()
+
+        role.users.append(user)
```
Comment (Dheavyman) [inline seeders/seed_user.py]: Nice work. The users have already been created with a role_id and center_id and don't require to be appended to role users or center users again. Please have a look.

Comment (GbengaOshinaga) [reply seeders/seed_user.py]: Will do

**Comment by brayoh** [comment]
@GbengaOshinaga Fix the merge conflicts

**Comment by GbengaOshinaga** [comment]
@brayoh Fixed

**Comment by nzediegwu1** [comment]
Great job! Just a little feedback on the postman documentation link. Presently it shows a html page full of json data when clicked. Since postman chrome plugin has been deprecated, you could consider using a markdown like this: [Run in Postman](https://app.getpostman.com/run-collection/16b6480fed8b2d7b3037), which would allow the user to choose postman desktop app for viewing the collection

**Comment by GbengaOshinaga** [comment]
@nzediegwu1 Actually, I believe the easier way is to copy the link and import it in your Postman. I reckon the link gives more flexibility as to how the collection is shared.

**Comment by SEUNAGBEYE** [comment]
@GbengaOshinaga Your commit message does not have a body. Please add the bodyl

**Comment by frankip** [comment]
Address the conflicting files

**Comment by SEUNAGBEYE** [comment]
@GbengaOshinaga, please get three approvals so this PR can be merged.

File: api/utilities/validators/person_validator.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    Parameters:
+        center_id(str): id of the center the person belongs to
+        person_id(str): id of the person
+
+    Returns:
+        person(object): person object if found
+    """
+
+    center = Center.get_or_404(center_id)
+    person = center.users.filter_by(id=person_id, deleted=False).first()
```
Comment (nzediegwu1) [inline api/utilities/validators/person_validator.py]:
```

File: api/utilities/validators/role_validator.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    Parameters:
+        role_data(str): role
+
+    Returns:
+        role(object): role object if found
+    """
+
+    role_data = RoleSchema().load_object_into_schema(role_data)
+    role = Role.query_().filter_by(**role_data).first()
```
Comment (nzediegwu1) [inline api/utilities/validators/role_validator.py]: HERE (line 22-28) looks similar to line 22 -28 of api/utilities/validators/person_validator.py module. You might wish to refactor like this:
``` return role or ValidationError({
                'message': serialization_errors['not_found'].format('Role')
            }, 404)
```

File: api/utilities/validators/role_validator.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    if not role:
+        raise ValidationError(
+            {
+                'message': serialization_errors['not_found'].format('Role')
+            }, 404)
+
+    return role
```
Comment (nzediegwu1) [inline api/utilities/validators/role_validator.py]: HERE (line 22-28) looks similar to line 22 -28 of api/utilities/validators/person_validator.py module. You might wish to refactor like this:
``` return role or ValidationError({
                'message': serialization_errors['not_found'].format('Role')
            }, 404)
```

File: api/views/user.py [hunk 6]
```diff
@@ -66,38 +101,33 @@ def post(self, center_id):
...
-            'status': 'success',
-            'message': SUCCESS_MESSAGES['user_added'].format(user.name, center.name),
-            'data': user_serialize_schema.dump(user).data
+        return ({
+            'status':
+            'success',
+            'message':
+            SUCCESS_MESSAGES['user_added'].format(user.name, center.name),
+            'data':
+            user_serialize_schema.dump(user).data
```
Comment (nzediegwu1) [inline api/views/user.py]: Nice work. You could consider formatting line 126 - 132 to be more readable by placing object and key on same line

**Pull request merged**